compact_activity_snapshot.proto

```diff
@@ -64,62 +64,67 @@ message Backend {
+    // Source: lwlock.h (BuiltinTrancheIds), lwlock.c (BuiltinTrancheNames)
+    WAIT_EVENT_LWTRANCHE_CLOG_BUFFERS = 146; // clog, XactBuffer
+    WAIT_EVENT_LWTRANCHE_COMMITTS_BUFFERS = 147; // commit_timestamp, CommitTSBuffer
+    WAIT_EVENT_LWTRANCHE_SUBTRANS_BUFFERS = 148; // subtrans, SubtransBuffer
+    WAIT_EVENT_LWTRANCHE_MXACTOFFSET_BUFFERS = 149; // multixact_offset, MultiXactOffsetBuffer
+    WAIT_EVENT_LWTRANCHE_MXACTMEMBER_BUFFERS = 150; // multixact_member, MultiXactMemberBuffer
+    WAIT_EVENT_LWTRANCHE_ASYNC_BUFFERS = 151; // async, NotifyBuffer
+    WAIT_EVENT_LWTRANCHE_OLDSERXID_BUFFERS = 152; // oldserxid, SerialBuffer
+    WAIT_EVENT_LWTRANCHE_WAL_INSERT = 153; // wal_insert, WALInsert
+    WAIT_EVENT_LWTRANCHE_BUFFER_CONTENT = 154; // buffer_content, BufferContent
+    WAIT_EVENT_LWTRANCHE_BUFFER_IO_IN_PROGRESS = 155; // buffer_io, BufferIO
+    WAIT_EVENT_LWTRANCHE_REPLICATION_ORIGIN = 156; // replication_origin, ReplicationOriginState
+    WAIT_EVENT_LWTRANCHE_REPLICATION_SLOT_IO_IN_PROGRESS = 157; // replication_slot_io, ReplicationSlotIO
+    WAIT_EVENT_LWTRANCHE_PROC = 158; // proc, LockFastPath
+    WAIT_EVENT_LWTRANCHE_BUFFER_MAPPING = 159; // buffer_mapping, BufferMapping
+    WAIT_EVENT_LWTRANCHE_LOCK_MANAGER = 160; // lock_manager, LockManager
+    WAIT_EVENT_LWTRANCHE_PREDICATE_LOCK_MANAGER = 161; // predicate_lock_manager, PredicateLockManager
+    WAIT_EVENT_LWTRANCHE_PARALLEL_HASH_JOIN = 162; // parallel_hash_join, ParallelHashJoin
+    WAIT_EVENT_LWTRANCHE_PARALLEL_QUERY_DSA = 163; // parallel_query_dsa, ParallelQueryDSA
+    WAIT_EVENT_LWTRANCHE_SESSION_DSA = 164; // session_dsa, PerSessionDSA
+    WAIT_EVENT_LWTRANCHE_SESSION_RECORD_TABLE = 165; // session_record_table, PerSessionRecordType
+    WAIT_EVENT_LWTRANCHE_SESSION_TYPMOD_TABLE = 166; // session_typmod_table, PerSessionRecordTypmod
+    WAIT_EVENT_LWTRANCHE_SHARED_TUPLESTORE = 167; // shared_tuplestore, SharedTupleStore
+    WAIT_EVENT_LWTRANCHE_TBM = 168; // tbm, SharedTidBitmap
+    WAIT_EVENT_LWTRANCHE_PARALLEL_APPEND = 169; // parallel_append, ParallelAppend
+    WAIT_EVENT_LWTRANCHE_PER_XACT_PREDICATE_LIST = 170; // serializable_xact, PerXactPredicateList
     // Source: lock.h (LockTagType enum), lockfuncs.c (LockTagTypeNames variable)
     WAIT_EVENT_LOCKTAG_RELATION = 200; // relation
     WAIT_EVENT_LOCKTAG_RELATION_EXTEND = 201;  // extend
     WAIT_EVENT_LOCKTAG_PAGE = 202; // page
     WAIT_EVENT_LOCKTAG_TUPLE = 203; // tuple
     WAIT_EVENT_LOCKTAG_TRANSACTION = 204; // transactionid
     WAIT_EVENT_LOCKTAG_VIRTUALTRANSACTION = 205; // virtualxid
-    WAIT_EVENT_LOCKTAG_SPECULATIVE_TOKEN = 206;  // speculative token
+    WAIT_EVENT_LOCKTAG_SPECULATIVE_TOKEN = 206;  // speculative token, spectoken
     WAIT_EVENT_LOCKTAG_OBJECT = 207; // object
     WAIT_EVENT_LOCKTAG_USERLOCK = 208; // userlock
     WAIT_EVENT_LOCKTAG_ADVISORY = 209; // advisory
+    WAIT_EVENT_LOCKTAG_FROZEN = 210; // frozenid
```

Comment:
[note for reviewer] related commit for a new lock: https://github.com/postgres/postgres/commit/566372b3d6435639e4cc4476d79b8505a0297c87